src/Data/List/NonEmpty2.agda

```diff
@@ -0,0 +1,170 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Alternative implementation of non-empty lists
+------------------------------------------------------------------------
+
+module Data.List.NonEmpty2 where
+
+open import Data.List.Base as List using (List ; [] ; _∷_)
+open import Data.Vec as Vec using (Vec ; [] ; _∷_)
+open import Data.Nat
+open import Data.Maybe using (Maybe ; just ; nothing)
+open import Data.Empty using (⊥)
+open import Data.Unit using (⊤ ; tt)
+open import Data.Product using (Σ ; _,_ ; proj₁ ; proj₂ ; ∃ ; ,_ ; _×_)
+
+open import Category.Monad
+open import Function
+
+not-empty : ∀ {a} {A : Set a} → List A → Set
```

Comment:
Propositions like this are usually capitalized, i.e. `Not-empty` or `NotEmpty`.

Reply:
thx I will modify this.

Reply:
`NotEmpty` is preferable.